index.js

```diff
@@ -385,6 +385,13 @@ Router.prototype.find = function find (method, path, version) {
       if (node === null) {
         return getWildcardNode(wildcardNode, method, originalPath, pathLenWildcard)
       }
+
+      // we need to know the outstanding path so far from the originalPath since the last encountered "/" and assign it to previousPath.
+      // e.g originalPath: /aa/bbb/cc, path: bb/cc
+      // outstanding path: /bbb/cc
+      const pathDiff = originalPath.slice(0, originalPathLength - pathLen)
+      const previousPath = pathDiff.slice(pathDiff.lastIndexOf('/') + 1, pathDiff.length) + path
```

Comment:
Can you check if a fast-path for this possible? I would prefer if we could detect the condition that lead to the issue being fixed by this PR, and then apply these checks instead?

Reply:
Good point, just added a check to verify the `previousPath` is "complete" since the last "/". 
This seems to be the condition as we keep trimming `path` which could lead to `previousPath` becoming one or more characters short.